File: CIVITAS/src/civitas/IrACarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class IrACarcel  extends Sorpresa{
    
    IrACarcel(Tablero tablero){
        super(tablero,-1,"Ir a carcel", null);
    }
    
   @Override
    void aplicarAJugador(int actual  ,ArrayList<Jugador> todos) {
        if(this.jugadorCorrecto(actual, todos)){
            this.informe(actual, todos);
            if(this.getTablero() != null)
                todos.get(actual).encarcelar(this.getTablero().getCarcel());
        }
    }
}

File: CIVITASRUBY/lib/ir_a_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  
  require_relative 'casilla'
  class Ir_a_carcel < Sorpresa
    def initialize(tablero,valor,texto)
      super(tablero,valor,texto,nil)
    end
    
    def aplicar_a_jugador(actual,todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        if @tablero != nil
          todos[actual].encarcelar(@tablero.get_carcel())
        end
      end
    end
  end
end
